compiler/plugins/target/AMD-AIE/aie/AIEAssignBufferAddressesBasic.cpp

```diff
@@ -0,0 +1,103 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "AIEAssignBufferAddressesBasic.h"
+
+#include "aie/Dialect/AIE/IR/AIEDialect.h"
+#include "llvm/ADT/Twine.h"
+#include "mlir/IR/Attributes.h"
+#include "mlir/Pass/Pass.h"
+
+#define DEBUG_TYPE "aie-assign-buffers-basic"
+
+using namespace mlir;
+using namespace xilinx;
+using namespace xilinx::AIE;
+
+struct AIEAssignBufferAddressesPassBasic : mlir::OperationPass<DeviceOp> {
+  MLIR_DEFINE_EXPLICIT_INTERNAL_INLINE_TYPE_ID(
+      AIEAssignBufferAddressesPassBasic)
+
+  AIEAssignBufferAddressesPassBasic()
+      : mlir::OperationPass<DeviceOp>(resolveTypeID()) {}
+  AIEAssignBufferAddressesPassBasic(
+      const AIEAssignBufferAddressesPassBasic &other)
+      : mlir::OperationPass<DeviceOp>(other) {}
+
+  llvm::StringRef getArgument() const override {
+    return "aie-assign-buffer-addresses-basic";
+  }
+
+  llvm::StringRef getName() const override {
+    return "AIEAssignBufferAddressesBasic";
+  }
+
+  std::unique_ptr<mlir::Pass> clonePass() const override {
+    return std::make_unique<AIEAssignBufferAddressesPassBasic>(
+        *static_cast<const AIEAssignBufferAddressesPassBasic *>(this));
+  }
+
+  void getDependentDialects(DialectRegistry &registry) const override {
+    registry.insert<func::FuncDialect>();
+    registry.insert<AIEDialect>();
+  }
+
+  void runOnOperation() override {
+    DeviceOp device = getOperation();
+    int counter = 0;
+    device.walk<WalkOrder::PreOrder>([&](BufferOp buffer) {
+      if (!buffer.hasName())
+        buffer.setSymName("_anonymous" + std::to_string(counter++));
+    });
+
+    for (auto tile : device.getOps<TileOp>()) {
+      const auto &targetModel = getTargetModel(tile);
+      int maxDataMemorySize = 0;
+      if (tile.isMemTile())
+        maxDataMemorySize = targetModel.getMemTileSize();
+      else
+        maxDataMemorySize = targetModel.getLocalMemorySize();
+      SmallVector<BufferOp, 4> buffers;
+      device.walk<WalkOrder::PreOrder>([&](BufferOp buffer) {
+        if (buffer.getTileOp() == tile) buffers.push_back(buffer);
+      });
+
+      std::sort(buffers.begin(), buffers.end(), [](BufferOp a, BufferOp b) {
+        return a.getAllocationSize() > b.getAllocationSize();
+      });
+
+      // Leave room at the bottom of the address range for stack
+      int stacksize = 0;
+      int address = 0;
+      if (auto core = tile.getCoreOp()) {
+        stacksize = core.getStackSize();
+        address += stacksize;
+      }
+
+      for (auto buffer : buffers) {
+        buffer.setAddress(address);
+        address += buffer.getAllocationSize();
+      }
+
+      if (address > maxDataMemorySize) {
+        InFlightDiagnostic error =
+            tile.emitOpError("allocated buffers exceeded available memory\n");
```

Comment:
```suggestion
            tile.emitOpError("allocated buffers exceeded available memory (") << address << ">" << maxDataMemorySize << ")\n";
```
and add a test for this. Example of 'negative' test: https://github.com/nod-ai/iree-amd-aie/blob/9773f55fc84ff0c12616da4d17719d362555893e/compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/test/pack_to_dma_failures.mlir#L4